qp_shotgun/humann2/humann2.py

```diff
@@ -137,21 +159,48 @@ def humann2(qclient, job_id, parameters, out_dir):
 
     # Step 3 execute humann2
-    commands_len = len(commands)
-    for i, cmd in enumerate(commands):
-        qclient.update_job_step(job_id, "Step 3 of 5: Executing HUMANn2"
-                                ", job %d/%d" % (i, commands_len))
-        std_out, std_err, return_value = system_call(cmd)
-        if return_value != 0:
-            error_msg = ("Error running HUMANn2:\nStd out: %s\nStd err: %s"
-                         % (std_out, std_err))
-            return False, None, error_msg
+    msg = "Step 3 of 5: Executing HUMANn2 job (%d/{0})".format(len(commands))
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    if not success:
+        return False, None, msg
 
     # Step 4 merge tables
-    qclient.update_job_step(job_id, "Step 4 of 5: Merging resulting tables")
-
-    # Step 5 generating re-normalized tables: TODO
-    qclient.update_job_step(job_id, "Step 5 of 5: Re-normalizing tables")
-    artifacts_info = []
-
-    return True, artifacts_info, ""
+    commands = []
+    commands.append(('humann2_join_tables -i {0} -o {0}/genefamilies.biom '
+                     '--file_name genefamilies --search-subdirectories '
+                     '--verbose').format(out_dir))
+    commands.append(('humann2_join_tables -i {0} -o {0}/pathcoverage.biom '
+                     '--file_name pathcoverage --search-subdirectories '
+                     '--verbose').format(out_dir))
+    commands.append(('humann2_join_tables -i {0} -o {0}/pathabundance.biom '
+                     '--file_name pathabundance --search-subdirectories '
+                     '--verbose').format(out_dir))
+    msg = "Step 4 of 5: Merging resulting tables job (%d/3)"
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    if not success:
+        return False, None, msg
+
+    # Step 5 generating re-normalized tables
+    commands = []
+    commands.append(('humann2_renorm_table -i {0}/genefamilies.biom -u cpm '
+                     '-o {0}/genefamilies_cpm.biom').format(out_dir))
+    commands.append(('humann2_renorm_table -i {0}/pathcoverage.biom -u relab '
+                     '-o {0}/pathcoverage_relab.biom').format(out_dir))
+    commands.append(('humann2_renorm_table -i {0}/pathabundance.biom -u relab '
+                     '-o {0}/pathabundance_relab.biom').format(out_dir))
+    msg = "Step 5 of 5: Re-normalizing tables (%d/3)"
+    success, msg = _run_commands(qclient, job_id, commands, msg)
+    if not success:
+        return False, None, msg
+
+    # Generating artifact
+    pb = partial(join, out_dir)
+    filepaths = [(pb('genefamilies.biom'), 'biom'),
+                 (pb('pathcoverage.biom'), 'biom'),
+                 (pb('pathabundance.biom'), 'biom'),
+                 (pb('genefamilies_cpm.biom'), 'biom'),
+                 (pb('pathcoverage_relab.biom'), 'biom'),
+                 (pb('pathabundance_relab.biom'), 'biom')]
+    ainfo = [ArtifactInfo('OTU table', 'BIOM', filepaths)]
```

Comment:
Thinking more about it (it just hit me now xD) I think is better to create 6 artifacts. The reason why is because in the analysis pipeline, you are expecting a biom table, so you want to be clear which biom do you want to use: normalized or not, and if it contains gene or path abundance/coverage